### PR TITLE
Added custom Huber loss

### PR DESCRIPTION
**Pull request by Vishal-V**
The Huber loss function describes the penalty incurred by an estimation procedure and the loss function is piecewise. This function is quadratic for small values of a, and linear for large values, with equal values and slopes of the different sections at the two points where the error is compared with the threshold.

### CODE REVIEW

**Comment by pavithrasv** [comment]
There is a Huber loss in keras losses already. If this is a different loss function may be it should be in the add ons repo?

**Comment by Vishal-V** [comment]
Okay @pavithrasv, I will add any other custom losses to the addons repository.

File: tensorflow/python/keras/losses.py [hunk 1]
```diff
@@ -433,6 +435,41 @@ def __init__(self,
...
+    step_error = y_true - y_pred
+    is_lesser = step_error < threshold
+    quadratic = square(error) / 2
+    linear = threshold * error - square(threshold) / 2
+    return array_ops.where(is_lesser, quadratic, linear)
+  return huber_fn
+
+
+@keras_export('keras.losses.HuberLoss')
+class HuberLoss(Loss):
```
Comment (ashutosh1919) [inline tensorflow/python/keras/losses.py]: Name of the function should not be `HuberLoss` since it is already acquired by the huber loss function.

File: tensorflow/python/keras/losses.py [hunk 2]
```diff
@@ -16,7 +16,7 @@
 """
 from __future__ import absolute_import
 from __future__ import division
-from __future__ import print_function
+from __future__ import printable_module_namefunction
```
Comment (ashutosh1919) [inline tensorflow/python/keras/losses.py]: Why did you change this one?

File: tensorflow/python/keras/losses.py [hunk 3]
```diff
@@ -38,6 +38,8 @@
 from tensorflow.python.ops.losses import util as tf_losses_util
 from tensorflow.python.util.tf_export import keras_export
 from tensorflow.tools.docs import doc_controls
+from tensorflow.math import reduce_sum, square
+from tensorflow import size, cast, float32
```
Comment (ashutosh1919) [inline tensorflow/python/keras/losses.py]: Note that you will import here considering tensorflow codebase and not API.  
In my opinion, these imports won't work. You need to import considering codebase here similar to other imports in the file.

**Review by ashutosh1919 — CHANGES_REQUESTED** [review]
Please provide link of the research paper or journal or article here for the reference.  
Also, if you are adding this loss in add ons repo, then please close this PR but consider suggested changes while making PR on add ons.

File: tensorflow/python/keras/losses.py (hunk 3 above)
Comment (Vishal-V) [reply tensorflow/python/keras/losses.py]: Ah yes. That was terrible from my part. I will change that.

File: tensorflow/python/keras/losses.py (hunk 1 above)
Comment (Vishal-V) [reply tensorflow/python/keras/losses.py]: Okay. I will change it and add some more checks to the code before adding it to addons.

File: tensorflow/python/keras/losses.py (hunk 2 above)
Comment (Vishal-V) [reply tensorflow/python/keras/losses.py]: This puzzled me as well when I raised the PR. I am positive I did not change it, I will revert it back nonetheless.
